src/Exthost/Exthost.rei

```diff
@@ -83,7 +83,11 @@ module Location: {
 };
 
 module MarkdownString: {
-  type t = string;
+  [@deriving show]
+  type t;
+
+  let fromString: string => t;
+  let toString: t => string;
```

Comment:
This might seem a bit convoluted, since the underlying type in `MarkdownString.re` is.... `string`.

But eventually there is going to be more we need to handle, like:
```
// https://github.com/onivim/vscode-exthost/blob/c7df89c1cf0087ca5decaf8f6d4c0fd0257a8b7a/src/vs/base/common/htmlContent.ts#L10
//type t = {
//	stringValue: string, // value
//	isTrusted: bool,
//	supportThemeIcons: bool,
// uris
//}
```

So if we make it abstract - it gives us more flexibility to rev the underlying implementation, without breaking any consumers.